File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/prof/TrancheHoraireProfCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.prof;

import java.util.List;
import ma.zs.alc.bean.core.prof.TrancheHoraireProf;
import ma.zs.alc.dao.criteria.core.prof.TrancheHoraireProfCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface TrancheHoraireProfCollaboratorService extends  IService<TrancheHoraireProf,TrancheHoraireProfCriteria>  {

    List<TrancheHoraireProf> findByProfId(Long id);
    int deleteByProfId(Long id);
    long countByProfRef(String ref);




}

File: frontend-bqckup/src/app/shared/service/collaborator/prof/TrancheHoraireProfCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {TrancheHoraireProfDto} from '../../../model/prof/TrancheHoraireProf.model';
import {TrancheHoraireProfCriteria} from '../../../criteria/prof/TrancheHoraireProfCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class TrancheHoraireProfCollaboratorService extends AbstractService<TrancheHoraireProfDto, TrancheHoraireProfCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/trancheHoraireProf/';
    }

    public constrcutDto(): TrancheHoraireProfDto {
        return new TrancheHoraireProfDto();
    }

    public constrcutCriteria(): TrancheHoraireProfCriteria {
        return new TrancheHoraireProfCriteria();
    }
}
